core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -415,9 +415,6 @@ Byte parse(Object value) {
                     if (doubleValue < Byte.MIN_VALUE || doubleValue > Byte.MAX_VALUE) {
                         throw new IllegalArgumentException("Value [" + value + "] is out of range for a byte");
                     }
-                    if (doubleValue % 1 != 0) {
-                        throw new IllegalArgumentException("Value [" + value + "] has a decimal part");
-                    }
```

Comment:
I'd still like an exception to be thrown if this is called from parseCreateField. Maybe add a `boolean coerce` parameter (consistent with the rest of this class) and only perform this check if `coerce` is `false`?